src/main/java/co/cask/gcp/publisher/GooglePublisher.java

```diff
@@ -66,8 +74,33 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   }
 
   @Override
-  public void prepareRun(BatchSinkContext context) {
+  public void prepareRun(BatchSinkContext context) throws IOException {
     config.validate();
+
+    TopicAdminSettings.Builder topicAdminSettings = TopicAdminSettings.newBuilder();
+    String serviceAccountPath = config.getServiceAccountFilePath();
+    if (serviceAccountPath != null) {
+      topicAdminSettings.setCredentialsProvider(() -> GCPUtils.loadServiceAccountCredentials(serviceAccountPath));
+    }
+    TopicAdminClient topicAdminClient = TopicAdminClient.create(topicAdminSettings.build());
+
+    String projectId = config.getProject();
+    ProjectTopicName projectTopicName = ProjectTopicName.of(projectId, config.topic);
+
+    try {
+      topicAdminClient.getTopic(projectTopicName);
+    } catch (NotFoundException e) {
```

Comment:
Is NotFoundException guaranteed to be  be thrown if it does not exists?

Reply:
from my experiments, this is how it behaves. Doesn't seem to return null.